lib/Alchemy/Phrasea/SearchEngine/Elastic/ElasticSearchEngine.php

```diff
@@ -292,8 +293,16 @@ public function query($string, $offset, $perPage, SearchEngineOptions $options =
             $results[] = ElasticsearchRecordHydrator::hydrate($hit, $n++);
         }
 
+        /** @var FacetsResponse $facets */
         $facets = $this->facetsResponseFactory->__invoke($res);
 
+        // for es, suggestions are a flat view of facets (api backward compatibility)
+        $suggestions->clear();
+        foreach($facets->getFacets() as $facet) {
+            foreach($facet['values'] as $value) {
+                $suggestions->add(new SearchEngineSuggestion($value['query'], $value['value'], $value['count']));
```

Comment:
This code should not be here, at least it should be extracted in another method or class, `ElasticSearchEngine` is too heavy already. Maybe this belong in some `FacetResponse::getSuggestions()` method.